src/main/java/practice/StreamPractice.java

```diff
@@ -75,6 +110,11 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * parametrized with Candidate in CandidateValidator.
      */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        return candidates
+                .stream()
+                .filter(c -> new CandidateValidator().validate(c))
```

Comment:
```suggestion
                .filter(new CandidateValidator())
```
it's enough just to pass an object of predicate type